versioned/spi/src/main/java/org/projectnessie/versioned/VersionStore.java

```diff
@@ -139,13 +143,16 @@ void merge(Hash fromHash, BranchName toBranch, Optional<Hash> expectedHash)
    * @param ref The named ref to be assigned
    * @param expectedHash The current head of the NamedRef to validate before updating. If not
    *     present, force assignment.
-   * @param targetHash The hash that this ref should refer to.
+   * @param target The named reference on which the commit to which {@code ref} shall be assigned to
+   *     exists.
+   * @param targetHash The hash that this ref should refer to, must be reachable from {@code
+   *     target}.
    * @throws ReferenceNotFoundException if {@code ref} is not present in the store or if {@code
    *     targetHash} is not present in the store
    * @throws ReferenceConflictException if {@code expectedHash} is not empty and its value doesn't
    *     match the stored hash for {@code ref}
    */
-  void assign(NamedRef ref, Optional<Hash> expectedHash, Hash targetHash)
```

Comment:
To me the targetHash and target are redundant. The hash of target should be targetHash. Why add this?